### ISSUE TITLE
No-unused-vars doesn't detect destructuring assignments

### ISSUE THREAD

**Issue by aneilbaboo** [eslint/eslint#8098]
**Tell us about your environment**

* **ESLint Version:**
v3.15.0
* **Node Version:**
v7.5.0
* **npm Version:**
4.1.2

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```
env:
  es6: true
  node: true
extends: 'eslint:recommended'
```

**What did you do? Please include the actual source code causing the issue.**

```js
var x;

[x] = [1];
```

**What did you expect to happen?**

This should validate without errors.

**What actually happened? Please include the actual, raw output from ESLint.**
```shell
$ eslint test.js
/Users/aneil/code/test/test.js
  1:5  error  'x' is assigned a value but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```
